src/components/ListItem.js

```diff
@@ -52,23 +53,37 @@ class ListItem extends Component {
   };
 
   state = {
-    editing: false
+    editing: false,
+    clonedSegment: {}
   };
 
   handleDelete = () => {
     this.props.deleteItem(this.props.item.id);
+    this.props.deleteSegment(this.props.item.id, this.props.smData);
```

Comment:
No big deal, but a generic rule of thumb is if you use a props variable more than once, destructure it.  ie. `const { id } = this.props.item;`